acme/client.go

```diff
@@ -100,6 +100,24 @@ func NewClient(caDirURL string, user User, keyBits int) (*Client, error) {
 	return &Client{directory: dir, user: user, jws: jws, keyBits: keyBits, solvers: solvers}, nil
 }
 
+// SetChallengeProvider specifies a custom provider that will make the solution available
+func (c *Client) SetChallengeProvider(challenge string, p ChallengeProvider) {
+	switch challenge {
+	case "http-01":
```

Comment:
I think when going down this route we should export constants for these identifiers. Like `acme.HTTP01`.